src/test/java/io/github/acm19/aws/interceptor/http/AwsRequestSigningApacheInterceptorTest.java

```diff
@@ -95,73 +99,52 @@ void signPostRequest() throws Exception {
 
         request.addHeader("foo", "bar");
 
-        HttpCoreContext context = new HttpCoreContext();
-        context.setTargetHost(HttpHost.create("localhost"));
-
-        interceptor.process(request, context);
+        client.execute(host, request);
+        RecordedRequest recorded = server.takeRequest();
 
-        assertEquals("bar", request.getFirstHeader("foo").getValue());
-        assertEquals("wuzzle", request.getFirstHeader("Signature").getValue());
-        assertEquals("required", request.getFirstHeader("x-amz-content-sha256").getValue());
+        assertEquals("bar", recorded.getHeader("foo"));
+        assertEquals("wuzzle", recorded.getHeader("Signature"));
+        assertEquals("required", recorded.getHeader("x-amz-content-sha256"));
 
         assertEquals(Long.toString(httpEntity.getContentLength()),
-                request.getFirstHeader("signedContentLength").getValue());
+                recorded.getHeader("signedContentLength"));
 
-        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-        request.getEntity().writeTo(outputStream);
-        assertEquals(payload, outputStream.toString());
+        assertEquals(payload, recorded.getBody().readUtf8());
     }
 
-    @Test
-    void testRepeatableEntity() throws Exception {
-        HttpEntityEnclosingRequest request = new BasicHttpEntityEnclosingRequest(
-                new MockRequestLine("POST", "/"));
-
-        String payload = "{\"test\": \"val\"}";
-        request.setEntity(new StringEntity(payload));
-        HttpCoreContext context = new HttpCoreContext();
-        context.setTargetHost(HttpHost.create("localhost"));
-        interceptor.process(request, context);
-
-        assertTrue(request.getEntity().isRepeatable());
-    }
```

Comment:
Dropped this test since the entity being repeatable seemed like an implementation detail that didn't matter so long as the requests are delivered in the proper format to the server.